nft-simple/src/internal.rs

```diff
@@ -157,3 +160,33 @@ impl Contract {
         (owner_id, approved_account_ids)
     }
 }
+
+pub(crate) fn convert_token_to_ext_object(token: Token) -> TokenReturnObject {
+    let mut json_map: HashMap<AccountId, U64> = HashMap::new();
+    for hm in token.approved_account_ids.clone() {
+        json_map.insert(hm.0, U64::from(hm.1));
+    }
+
+    TokenReturnObject {
+        owner_id: token.owner_id,
+        metadata: token.metadata,
+        approved_account_ids: json_map,
+        approval_counter: U64::from(token.approval_counter)
+    }
+}
+
+pub(crate) fn convert_hashmap_to_ext_object(map: HashMap<AccountId, u64>) -> HashMap<AccountId, U64> {
+    let mut json_map: HashMap<AccountId, U64> = HashMap::new();
+    for entry in map {
+        json_map.insert(entry.0, U64::from(entry.1));
+    }
+    json_map
+}
+
+pub(crate) fn convert_ext_hashmap_to_object(map: HashMap<AccountId, U64>) -> HashMap<AccountId, u64> {
+    let mut reg_map: HashMap<AccountId, u64> = HashMap::new();
+    for entry in map {
+        reg_map.insert(entry.0, entry.1.into());
+    }
+    reg_map
+}
```

Comment:
Why not store it as `HashMap<AccountId, U64>` internally?

Reply:
Well, I swear I had issues with Borsh being angry with me using U64, but I'm unable to reproduce it again.
I think this means I can sweep through and make a good amount of changes including removing the `TokenReturnObject` struct.